tests/DrupalConsoleStackTest.php

```diff
@@ -0,0 +1,141 @@
+        $this->assertEquals($expected, $command);
+    }
+
+    public function testConfigImportCommand()
+    {
+        $command = $this->taskDrupalConsoleStack()
+          ->directory('sites/default/config')
+          ->configImport()
+          ->getCommand();
+        $expected = 'drupal config:import --yes'
+          . ' --directory=' . escapeshellarg('sites/default/config');
+        $this->assertEquals($expected, $command);
+    }
+
+    public function testSiteInstallSqliteCommand()
+    {
+        $command = $this->taskDrupalConsoleStack()
+          ->siteName('Site Name')
+          ->siteMail('site-mail@example.com')
+          ->langcode('de')
+          ->accountMail('mail@example.com')
+          ->accountName('admin')
+          ->accountPass('pw')
+          ->dbPrefix('drupal_')
+          ->dbType('sqlite')
+          ->dbFile('sites/default/.ht.sqlite')
+          ->siteInstall('minimal')
+          ->getCommand();
+        $expected = 'drupal site:install minimal --yes --site-name=' . escapeshellarg('Site Name')
+          . ' --site-mail=site-mail@example.com'
+          . ' --langcode=de --account-mail=mail@example.com --account-name=' . escapeshellarg('admin')
+          . ' --account-pass=pw'
+          . ' --db-prefix=drupal_ --db-type=sqlite --db-file=' . escapeshellarg("sites/default/.ht.sqlite");
+        $this->assertEquals($expected, $command);
+    }
+
+    public function testSiteInstallMysqlCommand()
+    {
+        $command = $this->taskDrupalConsoleStack()
+          ->siteName('Site Name Mysql')
+          ->siteMail('site-mail2@example.com')
+          ->langcode('fr')
+          ->accountMail('mail2@example.com')
+          ->accountName('admin-user')
+          ->accountPass('passw')
+          ->dbPrefix('drupal_')
+          ->dbType('mysql')
+          ->dbHost('localhost')
+          ->dbName('testdb')
+          ->dbUser('dbuser')
+          ->dbPass('testdbpw')
+          ->siteInstall('standard')
+          ->getCommand();
+        $expected = 'drupal site:install standard --yes --site-name=' . escapeshellarg('Site Name Mysql')
+          . ' --site-mail=site-mail2@example.com --langcode=fr'
+          . ' --account-mail=mail2@example.com --account-name=' . escapeshellarg('admin-user')
+          . ' --account-pass=passw --db-prefix=drupal_ --db-type=mysql'
+          . ' --db-host=' . escapeshellarg('localhost') . ' --db-name=' . escapeshellarg('testdb')
+          . ' --db-user=' . escapeshellarg('dbuser') . ' --db-pass=' . escapeshellarg('testdbpw');
+        $this->assertEquals($expected, $command);
+    }
+
+    public function testDrupalConsoleList()
+    {
+        $result = $this->taskDrupalConsoleStack(__DIR__ . '/../vendor/bin/drupal')
+          ->printed(false)
+          ->listCommands()
+          ->run();
+        $this->assertEquals(trim($result->getMessage()),
+          'DrupalConsole must be executed within a Drupal Site.');
```

Comment:
You can execute `drupal list` outside of a Drupal site. See also [this comment](https://github.com/digipolisgent/robo-drupal-console/commit/d571633a276785f0cc61c2e0a57afd9b1ca49090#commitcomment-20181746).